src/pages/accommodations/AccommodationForm.tsx

```diff
@@ -0,0 +1,307 @@
+import { useUpdateAccommodation } from '@/api/mutations/accommodations/useUpdateAccommodation';
+import { useGetAccommodation } from '@/api/queries/accommodations/useGetAccommodation';
+import YandexMap from '@/components/YandexMap';
+import { ROUTES } from '@/config/routes.config';
+import { AccommodationReq, accommodationSchema } from '@/types/accommodation.types';
+
+export default function AccommodationForm() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { data: accommodation } = useGetAccommodation({ id });
+  const { mutate: createAccommodation } = useCreateAccommodation();
+  const { mutate: updateAccommodation } = useUpdateAccommodation();
+  const { mutate: deleteAccommodation } = useDeleteAccommodation();
+
+  const {
+    watch,
+    control,
+    setValue,
+    handleSubmit,
+    formState: { errors, dirtyFields },
+  } = useForm<AccommodationReq>({
+    resolver: zodResolver(accommodationSchema),
+    defaultValues: {
+      title: '',
+      description: '',
+      allowedNumberOfPeople: 1,
+      availableFrom: String(new Date()),
+      availableTo: String(new Date(dayjs().add(1, 'week').toDate())),
+      thumbnailUrl: '',
+      numberOfRooms: 1,
+      previewImgUrl: '',
+      price: 0,
+      squareMeters: 0,
+      address: {
+        street: '',
+        city: '',
+        country: '',
+        zipCode: '',
+        latitude: 0,
+        longitude: 0,
+      },
+    },
+    values: accommodation?.data,
+  });
+
+  const latitudeWatch = watch('address.latitude');
+  const longitudeWatch = watch('address.longitude');
+
+  const onSubmit = (data: AccommodationReq) => {
+    if (id) {
+      if (!Object.keys(dirtyFields).length) {
+        navigate(ROUTES.accommodations.root);
+        return;
+      }
+      updateAccommodation({
+        ...data,
+        id,
+        availableFrom: dayjs(data.availableFrom).toISOString(),
+        availableTo: dayjs(data.availableTo).toISOString(),
+      });
+      return;
+    }
+    createAccommodation({
+      ...data,
+      availableFrom: dayjs(data.availableFrom).toISOString(),
+      availableTo: dayjs(data.availableTo).toISOString(),
+    });
+  };
+
+  const handleDelete = () => {
+    if (id) {
+      deleteAccommodation(id);
+    }
+  };
+
+  const handleCoordsChange = (coords: [number, number]) => {
+    setValue('address.latitude', coords[0], {
+      shouldDirty: true,
+    });
+    setValue('address.longitude', coords[1], {
+      shouldDirty: true,
+    });
+  };
+
+  return (
+    <Box>
+      <Typography variant="h4" textAlign="center" pb={4}>
+        {id ? 'Edit' : 'Create'} Accommodation
+      </Typography>
+      <form onSubmit={(e) => void handleSubmit(onSubmit)(e)}>
+        <Box display={'grid'} gridTemplateColumns={'1fr 1fr'} gap={4} py={4}>
+          <Controller
+            name="title"
+            control={control}
+            render={({ field }) => (
```

Comment:
create memoized functions for each render. like renderTitle = useMemo((field)=>{
                <TextField
                {...field}
                label="Title"
                error={!!errors.title}
                helperText={errors.title?.message}
              />
}, [errors]